mkosi/install.py

```diff
@@ -47,11 +47,19 @@ def flock(path: Path) -> Iterator[Path]:
         os.close(fd)
 
 
-def copy_path(src: Path, dst: Path, preserve_owner: bool = True) -> None:
+def copy_path(
+    src: Path,
+    dst: Path,
+    *,
+    dereference: bool = False,
+    overwrite: bool = True,
+    preserve_owner: bool = True,
+) -> None:
     run([
         "cp",
         "--recursive",
-        "--no-dereference",
+        f"--{'' if dereference else 'no-'}dereference",
+        f"--update={'all' if overwrite else 'none'}",
```

Comment:
We should use --no-clobber here instead of --update

Reply:
So just to make sure, you want it to fail if `overwrite=False` and a file already exists?

Reply:
```console
$ touch foo
$ touch bar
$ cp --no-clobber foo bar
cp: not replacing 'bar'
$ echo $?
1
```

Reply:
This seems like a huge mess, let's do it in python instead, but keep it simple, just copy files, no need for a full recursive copy